packages/twenty-server/src/engine/api/graphql/workspace-schema-builder/type-definitions.generator.ts

```
@@ -263,11 +270,41 @@ export class TypeDefinitionsGenerator {
     // Generate extended object type defs only for objects that contain composite fields
     const objectMetadataCollectionWithCompositeFields =
       objectMetadataCollection.filter(objectContainsRelationField);
-    const objectTypeDefs = objectMetadataCollectionWithCompositeFields.map(
-      (objectMetadata) =>
-        this.extendObjectTypeDefinitionFactory.create(objectMetadata, options),
+    const workspaceId =
+      objectMetadataCollectionWithCompositeFields[0]?.workspaceId;
+
+    if (!workspaceId) {
+      throw new Error('Workspace ID not found');
```

we should throw a custom exception here?